pkg/controller/beat/common/pod_test.go

```diff
@@ -198,12 +311,73 @@ func assertConfiguration(t *testing.T, pod corev1.PodTemplateSpec) {
 			break
 		}
 	}
-	assert.NotNil(t, configVolume)
+	require.NotNil(t, configVolume)
 	// Validate the mode
-	assert.NotNil(t, configVolume.DefaultMode, "default volume mode for beat configuration should not be nil")
+	require.NotNil(t, configVolume.DefaultMode, "default volume mode for beat configuration should not be nil")
 	assert.Equal(t, expectedConfigVolumeMode, *configVolume.DefaultMode)
 }
 
+func assertMonitoring(t *testing.T, client k8s.Client, beat v1beta1.Beat, pod corev1.PodTemplateSpec) {
+	t.Helper()
+	var monitoringVolume *corev1.Volume
+	// Validate that the Pod's volumes contain a Secret as a monitoring CA volume.
+	for i := range pod.Spec.Volumes {
+		if pod.Spec.Volumes[i].Name == "beat-monitoring-certs" {
+			monitoringVolume = &pod.Spec.Volumes[i]
+			break
+		}
+	}
+	require.NotNil(t, monitoringVolume)
+	require.NotNil(t, monitoringVolume.Secret)
+	assert.Equal(t, monitoringVolume.Secret.SecretName, "testbeat-es-testes-ns-monitoring-ca")
+
+	if monitoring.IsMetricsDefined(&beat) {
+		// ensure the metricsbeat sidecar exists
+		assert.True(t, containersContains(pod.Spec.Containers, "metrics-monitoring-sidecar"))
+
+		// ensure that the beat's metrics port can be set, and used in the sidecar
+		var secret corev1.Secret
+		assert.NoError(t, client.Get(context.Background(), types.NamespacedName{Name: fmt.Sprintf("%s-beat-monitoring-metricbeat-config", beat.Name), Namespace: beat.Namespace}, &secret))
+		data, ok := secret.Data["metricbeat.yml"]
+		assert.True(t, ok)
+
+		var cfg metricbeatConfig
+		assert.NoError(t, yaml.Unmarshal(data, &cfg))
+		u, err := url.Parse(cfg.MetricBeat.Modules[0].Hosts[0])
+		require.NoError(t, err)
+		assert.Equal(t, "3033", u.Port())
+	}
+	if monitoring.IsLogsDefined(&beat) {
+		assert.True(t, containersContains(pod.Spec.Containers, "logs-monitoring-sidecar"))
```

Comment:
All these `assert.*` need to be `require`. `assert` returns a `bool` which is here simply ignored i.e. we would not even know if the test failed. `require` calls the `assert` variant and makes the test fail if the `assert` returns `false`.

Reply:
Turns out this is not true. Sorry about that. Test fails with `assert` just as well. TIL